### ISSUE TITLE
`--external` does not work for node browser polyfills

### ISSUE THREAD

**Issue by utftu** [oven-sh/bun#13941]
### What version of Bun is running?

1.1.27+267afa293

### What platform is your computer?

Darwin 23.6.0 arm64 arm

### What steps can reproduce the bug?

1. create file
```ts
// hello.ts
import { Readable } from 'node:stream'; // same behavior in case simple `from "stream"`

console.log'Readable', Readable);
```
2. run build command `bun build ./hello.ts`
3. get error `error: No matching export in "node:stream" for import "Readable"`
4. try externalize stream module - `bun build --external="node:stream" ./hello.ts` or `bun build --external="stream" ./hello.ts`
5. get same error

### What is the expected behavior?

`bun build --external="node:stream" ./hello.ts` externalize node:stream module

### What do you see instead?

no way to externalize only stream module

### Additional information

`bun build --external="*" ./hello.ts` works, but it externalize all deps. There is no way to do it only for a nodejs's module

**Comment by Jarred-Sumner** [comment]
This doesn't answer your question, but if you want to build for Node you can do `--target=node` like `bun build --target=node ./hello.ts`

**Comment by 2007heaven** [comment]
allowing making fs external unblocks nwjs users from very fast builds.

**Comment by robobun** [comment]
Same bug as #2701: with the default browser target, `bun build --external <node builtin>` (crypto there, node:stream here) is ignored and the bundler still substitutes its node builtin browser polyfill instead of leaving the import external. The fix in #35212 (check user externals before applying browser node polyfills) covers both issues, so tracking continues in #2701.

Duplicate of #2701

Please add a 👍 or any extra details over there so it all stays in one place. If this is actually a different problem, say so here and we'll reopen.
